ChangeLog.d/8824.txt

```diff
@@ -0,0 +1,6 @@
+Bugfix
+   * The functions mbedtls_pk_sign(), mbedtls_pk_verify(), mbedtls_pk_decrypt()
+     or mbedtls_pk_encrypt() on a context holding an RSA key that was configured
+     to use PCKS#1 v2.1 (PSS or OAEP), for example by previously calling
+     mbedtls_rsa_set_padding(mbedtls_pk_rsa(...), MBEDTLS_RSA_PKCS1_V21, ...),
+     were incorrectly using PKCS#1 v1.5 when MBEDTLS_USE_PSA_CRYPTO was enabled.
```

Comment:
Non-blocker, but the wording of this entry bothers me a little. It says “this was wrong before”, but it doesn't explicitly say “we've fixed it”. We normally word changelog entries to say “this is the new behavior” as well as “this is why the previous behavior was wrong”.

Also we normally indicate the issue (although I only consider that a blocker when it's user-reported).

So I'd prefer something like

> Fix mbedtls_pk_sign(), mbedtls_pk_verify(), mbedtls_pk_decrypt() or mbedtls_pk_encrypt() on non-opaque RSA keys to honor the padding mode in the RSA context. Before they always used PKCS#1 v1.5 even when the RSA context was configured for PKCS#1 v2.1 (PSS/OAEP). Fixes #8824.